app/services/mapping/goodsshipment/GoodsShipmentBuilder.scala

```diff
@@ -16,12 +16,41 @@
 
 package services.mapping.goodsshipment
 import forms.Choice
+import javax.inject.Inject
 import services.cache.ExportsCacheModel
 import services.mapping.goodsshipment.consignment.ConsignmentBuilder
 import services.mapping.governmentagencygoodsitem.GovernmentAgencyGoodsItemBuilder
 import uk.gov.hmrc.http.cache.client.CacheMap
+import wco.datamodel.wco.dec_dms._2.Declaration
 import wco.datamodel.wco.dec_dms._2.Declaration.GoodsShipment
 
+class GoodsShipmentBuilder @Inject()(
```

Comment:
Here too, is this trait important?

Reply:
It doesn't offer us much at the moment, but having a trait makes it much easier for us to refactor the builders in the future

e.g. if we want to change the signature of `buildThenAdd` to be more like `def build(model: ExportsCacheModel, declaration: Declaration): Declaration` for example, all we would have to do is stick method in the trait, no more work required

Reply:
YAGNI & KISS

Reply:
will change

Reply:
> YAGNI & KISS

I think when 20+ classes have exactly the same signature though it shouts out that an interface should exist. Even if *at first* it doesn't add much value